### ISSUE TITLE
Add debug logging

### ISSUE THREAD

**Issue by sdebruyn** [microsoft/terraform-provider-azuredevops#381]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add more logging about which HTTP requests and responses are being sent. When you set the logging level to TRACE, you don't see anything at all at the moment.

I'm experiencing issues where I get HTTP 502 errors with this provider and I want to look into it, but I can't easily do that without proper logging.

### New or Affected Resource(s)

all resources

### Potential Terraform Configuration

TF_LOG=TRACE

### References

https://www.terraform.io/docs/internals/debugging.html

**Comment by xuzhang3** [comment]
Hi @sdebruyn The real HTTP response is processed in the azuredevops_sdk,  the response we get has been parsed.  We can add a switch for the debug in ADO provider, but you may not be able get the data you want.  If you are familiar with Golang and Terraform provider development, you can debug the provider source code directly.

**Comment by stewartadam** [comment]
Agreed, there needs to be user-accessible debugging. The Azure RM provider does this well, showing the REST API request and response pairs so you can get more info than just the HTTP response code when a failure occurs.

**Comment by xuzhang3** [comment]
@stewartadam Can you be more specific or give an example? If you have some good ideas, you can submit a PR for this issue.

**Comment by stewartadam** [comment]
#398 describes one scenario where request/resposne pair debugging is necessary. If you're getting 4xx errors, presently even with `TF_LOG=trace` the azure devops provider doesn't provide any additional information, making it very difficult to identify the errors.

Desired/expected behavior would be to see the HTTP request/response being sent to the API with TF_LOG=trace so that we can identify which endpoint is failing, and why, instead of just the status code (as is done when using the AzureRM provider).

**Comment by xuzhang3** [comment]
@stewartadam  Thanks  for your suggestions. Most ADO error log only print the service error with limited informations. From your suggestions,  ADO  error log should include at least the identifier's like `project-subresource1-subresource2` and context meta data when error occur like resp/req data or other data which may helpful.

**Comment by stewartadam** [comment]
Thx, agreed uri/request/response would be helpful to identify which call was causing the 4xx errors.
